### ISSUE TITLE
Consumer plugins are unsupported

### ISSUE THREAD

**Issue by tophercullen** [mybuilder/kongfig#70]
I found this issue while Trying to configure the rate-limiting plugin various ways. See example config below

```
apis:
  - name: someapi 
    ensure: "present" 
    attributes: .....
    plugins:
      - name: rate-limiting 
        ensure: "present" 
        attributes: 
          consumer_id: ID1
          config:
            minute: 1
      - name: rate-limiting 
        ensure: "present" 
        attributes: 
          consumer_id: ID2
          config:
            minute: 2`

plugins:
  - name: rate-limiting
    ensure: "present"
    attributes:
      consumer_id: ID1
      config:
        minute: 10
```

The result after running is something as follows:

```    
{
      "consumer_id": "ID1",
      "id": "someUUID",
      "enabled": true,
      "name": "rate-limiting"
      ....
         minute: 10
.     ....
    },
    {
      "consumer_id": "ID2",
      "api_id": "myapiID",
      "id": "someUUID",
      "enabled": true,
      "name": "rate-limiting",
      ......
         minute: 2
      ......
    }
```

This is not the desired outcome. Instead of two plugins being created for ID1, one with only an consumer_id and another with both consumer and api IDs, there is a singular plugin with a consumer_id. Meaning the global plugin config took precedence. 

Further subsequent runs with the same config fails:

```
POST 409 http://kong:8001/plugins 
 { consumer_id: 'ID1',
  config: { minute: 10 },
  name: 'rate-limiting' }
Error: Conflict
{"name":"already exists with value 'rate-limiting'"}
```

Similarly, If a global plugin exists, without a customer_id, it will succeed the first time, by converting the global plugin to one bound to a consumer_id. Subsequent runs will fail for a similar reason

It shouldn't come as a surprise that creating a generic global plugin again, then running the config again results it in being converted, and having duplicate identical global plugins which are essentially invisible to kongfig

At least for the global plugin, the core reason seems to be the global plugin discovery explicitly filters out plugin results that have a consumer or api ID (see https://github.com/mybuilder/kongfig/blob/master/src/kongState.js#L45)

**Comment by sarraz1** [comment]
Hello,
You right, i have reproduced the problems.

- With global plugin, when we declare 2 rate-limiting plugins for 2 différent consumers, kongfig considers the second declaration is a repetition of the first and raises a conflict error.
- With api plugins,  when we declare 2 rate-limiting plugins for 2 différent consumers, kongfig considers the second plugin is a repetition of the first and so, applying the second plugin will update the first(PATCH request)

Now I'm going to test your fix, I keep you inform soon.

**Comment by CyExy** [comment]
Closed by #80
